lib/util/BrowserWebsocket.js

```diff
@@ -0,0 +1,62 @@
+let EventEmitter;
+try {
+    EventEmitter = require("eventemitter3");
+} catch(err) {
+    EventEmitter = require("events").EventEmitter;
+}
+
+/**
+* Represents a browser's websocket usable by Eris
+* @extends EventEmitter
+* @prop {string} url The URL to connect to
+*/
+class BrowserWebsocket extends EventEmitter {
+    constructor(url) {
+        super();
+
+        if(typeof window === "undefined") {
+            new Error("BrowserWebsocket cannot be used outside of a browser environment");
+        }
+
+        this._ws = new window.WebSocket(url);
+        this._ws.onopen = () => this.emit("open");
+        this._ws.onmessage = this._onMessage.bind(this);
+        this._ws.onerror = (event) => this.emit("error", event);
```

Comment:
`error` event should be an actual `Error` object. I forget what info the arg has, but a generic `Error` would suffice if it doesn't have useful info